frontend/src/concepts/trustyai/utils.ts

```diff
@@ -15,3 +23,60 @@ export const formatListResponse = (x: BaseMetricListResponse): BiasMetricConfig[
     thresholdDelta: m.request.thresholdDelta,
     unprivilegedAttribute: m.request.unprivilegedAttribute.value,
   }));
+
+export const getTrustyStatusState = (
+  crFetchState: FetchState<TrustyAIKind | null>,
+  successDetails?: UseTrustyBrowserStorage,
+): TrustyStatusStates => {
+  const [cr, loaded, error] = crFetchState;
+
+  if (error) {
+    return { type: TrustyInstallState.ERROR, message: error.message };
+  }
+
+  if (!loaded) {
+    return { type: TrustyInstallState.LOADING_INITIAL_STATE };
+  }
+
+  if (!cr) {
+    // No CR, uninstalled
+    return { type: TrustyInstallState.UNINSTALLED };
+  }
+
+  if (cr.metadata.deletionTimestamp) {
+    return { type: TrustyInstallState.UNINSTALLING };
+  }
+
+  // Have a CR, getting specific state
+  const availableCondition = getConditionForType(cr, 'Available');
+  if (availableCondition?.status === 'True' && cr.status?.phase === 'Ready') {
+    // Installed and good to go
+    return {
+      type: TrustyInstallState.INSTALLED,
+      showSuccess: !!successDetails?.showSuccess,
+      onDismissSuccess: successDetails?.onDismissSuccess,
+    };
+  }
+
+  const dbAvailableCondition = getConditionForType(cr, 'DBAvailable');
+  if (dbAvailableCondition?.status === 'False') {
+    // Some sort of DB error -- try to show specifically what it is
+    return {
+      type: TrustyInstallState.ERROR,
+      message: `${dbAvailableCondition.reason ?? 'Unknown reason'}: ${
+        dbAvailableCondition.message ?? 'Unknown error'
+      }`,
+    };
+  }
+
+  if (availableCondition?.status === 'False') {
+    // Try to present the generic error as one last fallback
+    return {
+      type: TrustyInstallState.ERROR,
+      message: availableCondition.message ?? availableCondition.reason ?? 'Unknown available error',
+    };
+  }
```

Comment:
This logic needs to be updated to fit a better model with reading the CR -- waiting on feedback from Trusty Backend team

Reply:
Backend is working on supplying better conditional statuses -- hopefully this helps out. We may need to improve this code if the status cannot be provided in a direct accessible way

Reply:
right now it basically says there is a database error which makes me think its done loading... i guess this is on the backend to improve and not the UI?

Reply:
ok it flips between success and failed when the cr always says not ready?

Reply:
this doesnt really matter i guess as long as it shows the state is what this PR looks at